x-pack/platform/plugins/shared/triggers_actions_ui/public/application/hooks/use_rule_template.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+
+import { useKibana } from '../../common/lib/kibana';
+import { loadRuleTemplate } from '../lib/rule_template_api/get_rule_template';
+
+export interface UseRuleTemplateProps {
+  templateId?: string;
+  enabled: boolean;
+}
+
+export function useRuleTemplate(props: UseRuleTemplateProps) {
+  const { enabled, templateId } = props;
+  const { http } = useKibana().services;
+
+  const queryFn = () => {
+    if (!templateId) {
+      throw new Error('templateId not defined');
+    }
+    return loadRuleTemplate({ http, templateId });
+  };
+
+  const { data, error, isFetching, isError, isLoadingError, isLoading } = useQuery({
+    queryKey: ['getQueryDelaySettings'],
```

Comment:
```suggestion
    queryKey: ['getRuleTemplate', templateId],
```

Reply:
I noticed that the request is being made twice. Maybe due to the `queryKey`.

<img width="1069" height="363" alt="Screenshot 2025-09-17 at 2 42 59 PM" src="https://github.com/user-attachments/assets/47cbd4d6-56d2-4a6c-96fd-0d517901b8a9" />